homeassistant/components/samsungtv/config_flow.py

```diff
@@ -0,0 +1,197 @@
+        for entry in hass.config_entries.async_entries(DOMAIN)
+    )
+
+
+def _get_ip(host):
+    if host is None:
+        return None
+    return socket.gethostbyname(host)
+
+
+def _try_connect(host, name):
+    """Try to connect and check auth."""
+    for method in METHODS:
+        config = {
+            "name": "HomeAssistant",
+            "description": name,
+            "id": "ha.component.samsung",
+            "host": host,
+            "method": method,
+            "port": None,
+            "timeout": None,
+        }
+        try:
+            LOGGER.debug("Try config: %s", config)
+            with Remote(config.copy()):
+                LOGGER.debug("Working config: %s", config)
+                return RESULT_SUCCESS
+        except AccessDenied:
+            LOGGER.debug("Working but denied config: %s", config)
+            return RESULT_AUTH_MISSING
+        except UnhandledResponse:
+            LOGGER.debug("Working but unsupported config: %s", config)
+            return RESULT_NOT_SUPPORTED
+        except (OSError):
+            LOGGER.debug("Failing config: %s", config)
+
+    LOGGER.debug("No working config found")
+    return RESULT_NOT_FOUND
+
+
+class SamsungTVConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Samsung TV config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+
+    def __init__(self):
+        """Initialize flow."""
+        self._broadcast = None
+        self._host = None
+        self._ip = None
+        self._mac = None
+        self._manufacturer = None
+        self._model = None
+        self._name = None
+        self._port = None
+        self._timeout = None
+        self._title = None
+        self._uuid = None
+
+    def _get_entry(self):
+        return self.async_create_entry(
+            title=self._title,
+            data={
+                CONF_BROADCAST_ADDRESS: self._broadcast,
+                CONF_HOST: self._host,
+                CONF_ID: self._uuid,
+                CONF_IP_ADDRESS: self._ip,
+                CONF_MAC: self._mac,
+                CONF_MANUFACTURER: self._manufacturer,
+                CONF_MODEL: self._model,
+                CONF_NAME: self._name,
+                CONF_PORT: self._port,
+                CONF_TIMEOUT: self._timeout,
+            },
+        )
+
+    async def async_step_import(self, user_input=None):
+        """Handle configuration by yaml file."""
+        return await self.async_step_user(user_input)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if user_input is not None:
+            ip_address = await self.hass.async_add_executor_job(
+                _get_ip, user_input[CONF_HOST]
+            )
+
+            if _is_already_in_progress(self._async_in_progress(), ip_address):
```

Comment:
We may be able to replace these checks by setting a unique id for the config entry. See the hue config flow for example.

Reply:
The only unique identifier which is available in all steps would be the IP. So I could use this as `unique_id`.